# TestIndexErrors

> Auto-generated documentation for [env.Lib.site-packages.numpy.core.tests.test_indexerrors](..\..\..\..\..\..\..\env\Lib\site-packages\numpy\core\tests\test_indexerrors.py) module.

- [Smart-house-rest-api](..\..\..\..\..\..\README.md#description) / [Modules](..\..\..\..\..\..\MODULES.md#smart-house-rest-api-modules) / `Env` / `Lib` / `Site-packages` / [Numpy](..\..\index.md#numpy) / [Core](..\index.md#core) / [Tests](index.md#tests) / TestIndexErrors
    - [TestIndexErrors](#testindexerrors)
        - [TestIndexErrors().test_arraytypes_fasttake](#testindexerrorstest_arraytypes_fasttake)
        - [TestIndexErrors().test_iterators_exceptions](#testindexerrorstest_iterators_exceptions)
        - [TestIndexErrors().test_mapping](#testindexerrorstest_mapping)
        - [TestIndexErrors().test_mapping_error_message](#testindexerrorstest_mapping_error_message)
        - [TestIndexErrors().test_methods](#testindexerrorstest_methods)
        - [TestIndexErrors().test_multiindex_exceptions](#testindexerrorstest_multiindex_exceptions)
        - [TestIndexErrors().test_put_exceptions](#testindexerrorstest_put_exceptions)
        - [TestIndexErrors().test_take_from_object](#testindexerrorstest_take_from_object)

## TestIndexErrors

[[find in source code]](..\..\..\..\..\..\..\env\Lib\site-packages\numpy\core\tests\test_indexerrors.py#L7)

```python
class TestIndexErrors():
```

Tests to exercise indexerrors not covered by other tests.

### TestIndexErrors().test_arraytypes_fasttake

[[find in source code]](..\..\..\..\..\..\..\env\Lib\site-packages\numpy\core\tests\test_indexerrors.py#L10)

```python
def test_arraytypes_fasttake():
```

take from a 0-length dimension

### TestIndexErrors().test_iterators_exceptions

[[find in source code]](..\..\..\..\..\..\..\env\Lib\site-packages\numpy\core\tests\test_indexerrors.py#L52)

```python
def test_iterators_exceptions():
```

cases in iterators.c

### TestIndexErrors().test_mapping

[[find in source code]](..\..\..\..\..\..\..\env\Lib\site-packages\numpy\core\tests\test_indexerrors.py#L88)

```python
def test_mapping():
```

cases from mapping.c

### TestIndexErrors().test_mapping_error_message

[[find in source code]](..\..\..\..\..\..\..\env\Lib\site-packages\numpy\core\tests\test_indexerrors.py#L116)

```python
def test_mapping_error_message():
```

### TestIndexErrors().test_methods

[[find in source code]](..\..\..\..\..\..\..\env\Lib\site-packages\numpy\core\tests\test_indexerrors.py#L125)

```python
def test_methods():
```

cases from methods.c

### TestIndexErrors().test_multiindex_exceptions

[[find in source code]](..\..\..\..\..\..\..\env\Lib\site-packages\numpy\core\tests\test_indexerrors.py#L31)

```python
def test_multiindex_exceptions():
```

### TestIndexErrors().test_put_exceptions

[[find in source code]](..\..\..\..\..\..\..\env\Lib\site-packages\numpy\core\tests\test_indexerrors.py#L42)

```python
def test_put_exceptions():
```

### TestIndexErrors().test_take_from_object

[[find in source code]](..\..\..\..\..\..\..\env\Lib\site-packages\numpy\core\tests\test_indexerrors.py#L18)

```python
def test_take_from_object():
```
